src/components/Assemblies/CabinetRole.svelte

```diff
@@ -1,21 +1,56 @@
 <script lang="ts">
-	import type { CabinetSeat } from './shared';
+	import type { AssemblyMember } from '$lib/datasheets/assembly-member';
+	import type { CabinetSeat, TooltipProp } from './shared';
 
 	export let cabinets: CabinetSeat[];
 	export let role: CabinetSeat['role'];
+
+	let tooltipProp: TooltipProp | null = null;
+
+	const showTooltip = (event: MouseEvent, member: AssemblyMember) => {
+		const name = `${member.firstname} ${member.lastname}`;
+
+		tooltipProp = {
+			title: name,
+			additional: member.assemblyRole.role,
+			x: event.layerX,
+			y: event.layerY + 20
+		};
+	};
+
+	const hideTooltip = () => {
+		tooltipProp = null;
+	};
 </script>
 
 <div class="role">
 	<p class="heading-compact-01">{role}</p>
 	<div class="group-dot">
 		{#each cabinets.find((c) => role === c.role)?.parties || [] as party}
-			{#each Array.from({ length: party.count || 0 }, (_, i) => i) as _}
-				<span class="dot" style="background-color: {party.color || '#8D8D8D'};"></span>
+			{#each party?.members ?? [] as member}
+				<div
+					role="tooltip"
+					class="dot"
+					style="background-color: {party.color || '#8D8D8D'};"
+					on:mouseenter={(e) => showTooltip(e, member)}
+					on:mouseleave={hideTooltip}
+				></div>
 			{/each}
 		{/each}
 	</div>
 </div>
 
+<div
+	style:left={`${tooltipProp?.x ?? 0}px`}
+	style:top={`${tooltipProp?.y ?? 0}px`}
+	class:hidden={!tooltipProp}
```

Comment:
I wonder are there any reasons you use CSS hidden over [Svelte's `#if` block](https://svelte.dev/docs/logic-blocks#if)?

Because Svelte's `#if` block:
- Won't render the element from the DOM tree instead of just hiding it with CSS = no unused element in the DOM
- Inside the if block, the Svelte compiler will understand that the condition must be true. For example if you use `{#if tooltipProp !== null}` inside the block you can access the property without using optional changing at all eg `tooltipProp.title` is working because Svelte knows that tooltipProp must be not null to render this block.

If you have used other frameworks before, Svelte's `#if` is like Vue's `v-if` or React `{condition ? <element /> : null}` which is preferable than CSS's hidden if it fits the use case.

Reply:
I've tried before I use directive to set element's position and it was not working with position. It seem works now.

Thank for pointing me out.